multicluster/controllers/multicluster/commonarea/labelidentityimport_controller.go

```diff
@@ -0,0 +1,167 @@
+	var labelIdentityImport multiclusterv1alpha1.LabelIdentityImport
+	err := r.remoteCommonArea.Get(ctx, req.NamespacedName, &labelIdentityImport)
+	isDeleted := apierrors.IsNotFound(err)
+	if err != nil {
+		if !isDeleted {
+			klog.InfoS("Unable to fetch LabelIdentityImport", "labelidentityimport", req.NamespacedName.String(), "err", err)
+			return ctrl.Result{}, err
+		}
+		labelIdentityImportObj, exist, _ := r.installedLabelImports.GetByKey(req.NamespacedName.String())
+		if !exist {
+			// Stale controller will clean up any stale Label Identities
+			klog.ErrorS(err, "No cached data for LabelIdentityImport", "labelidentityimport", req.NamespacedName)
+			return ctrl.Result{}, nil
+		}
+		labelIdentityImport = labelIdentityImportObj.(multiclusterv1alpha1.LabelIdentityImport)
+		return r.handleLabelIdentityImpDelete(ctx, &labelIdentityImport)
+	}
+	return r.handleLabelIdentityImpUpdate(ctx, &labelIdentityImport)
+}
+
+func (r *LabelIdentityImportReconciler) handleLabelIdentityImpUpdate(ctx context.Context,
+	labelImp *multiclusterv1alpha1.LabelIdentityImport) (ctrl.Result, error) {
+	labelIdentityName := types.NamespacedName{
+		Namespace: "",
+		Name:      labelImp.Name,
+	}
+	labelIdentity := &multiclusterv1alpha1.LabelIdentity{}
+	err := r.localClusterClient.Get(ctx, labelIdentityName, labelIdentity)
+	labelIdentityNotFound := apierrors.IsNotFound(err)
+	if err != nil && !labelIdentityNotFound {
+		return ctrl.Result{}, err
+	}
+	if labelIdentityNotFound {
+		newLabelIdentity := &multiclusterv1alpha1.LabelIdentity{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: labelImp.Name,
+			},
+			Spec: multiclusterv1alpha1.LabelIdentityImportSpec{
+				Label: labelImp.Spec.Label,
+				ID:    labelImp.Spec.ID,
+			},
+		}
+		if err = r.localClusterClient.Create(ctx, newLabelIdentity, &client.CreateOptions{}); err != nil {
+			klog.ErrorS(err, "Failed to create LabelIdentity", "clusterID", r.localClusterID, "label", labelImp.Spec.Label)
+			return ctrl.Result{}, err
+		}
+		r.installedLabelImports.Add(*labelImp)
+	} else {
+		// TODO: is this branch necessary?
+		if labelIdentity.Spec.ID != labelImp.Spec.ID {
+			labelIdentity.Spec.ID = labelImp.Spec.ID
+			if err = r.localClusterClient.Update(ctx, labelIdentity, &client.UpdateOptions{}); err != nil {
+				klog.ErrorS(err, "Failed to update LabelIdentity", "clusterID", r.localClusterID, "label", labelImp.Spec.Label)
+				return ctrl.Result{}, err
+			}
+		}
+	}
+	return ctrl.Result{}, nil
+}
+
+func (r *LabelIdentityImportReconciler) handleLabelIdentityImpDelete(ctx context.Context,
+	labelImp *multiclusterv1alpha1.LabelIdentityImport) (ctrl.Result, error) {
+	labelIdentityName := types.NamespacedName{
+		Namespace: "",
+		Name:      labelImp.Name,
+	}
+	klog.V(2).InfoS("Deleting LabelIdentity corresponding to LabelIdentityImport", "labelidentityimport", klog.KObj(labelImp))
+	labelIdentity := &multiclusterv1alpha1.LabelIdentity{}
+	err := r.localClusterClient.Get(ctx, labelIdentityName, labelIdentity)
```

Comment:
Why do you need to get the `LabelIdentity` before delete it?  I feel it's OK to delete it directly without checking it.

Reply:
I agree that we could delete it without getting it first, but this seems to be the current pattern in all other places:
```
svcImp := &k8smcsv1alpha1.ServiceImport{}
err = r.localClusterClient.Get(ctx, svcImpName, svcImp)
if err != nil {
	return ctrl.Result{}, client.IgnoreNotFound(err)
}
err = r.localClusterClient.Delete(ctx, svcImp, &client.DeleteOptions{})
```
I'm not sure if it is because the `Delete` interface requires the object to be filled with resourceVersion etc.

Reply:
@GraysonWu @Dyanngg  Could refine this part and delete the labelIdentity directly without checking? I will check and update other kinds of resources.